### ISSUE TITLE
Autocomplete CSS class/id in HTML/PHP files (and vice versa optionally)

### ISSUE THREAD

**Issue by PixelT** [atom/autocomplete-html#13]
It will be nice, when I write:
```
<div class="">
```
and the cursor is betwteen class="|" the autocomplete will show the class / id which I defined in my sass / scss / less / css files

Optionally: When I start write in sass / scss / less / css files by typing . or # the autocomplete display class / id which I defined in my HTML / PHP files.

**Comment by florianbouvot** [comment]
+1

**Comment by sadovnychyi** [comment]
Take a look on my PL!

**Comment by PixelT** [comment]
Somebody think / work about this functionality?

**Comment by sadovnychyi** [comment]
@PixelT https://github.com/atom/autocomplete-html/pull/21 – it works, but still needs a lot of improvements.

**Comment by admosity** [comment]
This has been the crux of my life since I switched to atom. I work a lot in angular and none of the word completions are available to me.

I can switch autocomplete-plus' default provider to fuzzy search instead of symbol, but then I don't get the nice symbol notifications:

![image](https://cloud.githubusercontent.com/assets/4655972/15016476/51d63998-11df-11e6-81d7-8709165e3f4d.png)

Fuzzy (yes! completions):

![image](https://cloud.githubusercontent.com/assets/4655972/15016513/8281c404-11df-11e6-8409-d7f88d86dacc.png)

Symbol (no auto completions 😦 ):

![image](https://cloud.githubusercontent.com/assets/4655972/15016537/ab211324-11df-11e6-83e8-5a5ab8589691.png)

I am pretty new to atom plugin development, but I've done some PRs to some linters in atom. If someone can point me to some docs and where to start to help out; I'd gladly contribute!

**Comment by onetrev** [comment]
I was super bummed when I saw recently that Atom 1.18 beta listed this in the improvements:
_"Improved suggestions in autocomplete-html"_

But sadly, that did not include any movement on this issue. Am I in the minority here thinking the lack of this functionality is a **major** hinderance to anyone doing web dev with Atom, which has to be a big percentage of the users?

**Comment by 50Wliu** [comment]
Yep I'm aware.  As you can see by #21, integrating CSS completions properly is a much harder task to do than the improvements I made in Atom 1.18.  No one has had time to review that PR yet.

**Comment by onetrev** [comment]
Cool, sorry to pester. Just good to know it's still on the radar -- thank you! I did see this [sass styles autocomplete](https://github.com/atom/autocomplete-css/issues/98) added to autocomplete-css. That is awesome, as it solved 50% of the issue (well if you are a sass user that is) of having to always have all your files / tabs open all the time.

I don't suppose the code in there would be helpful here hey, since it's only just sourcing out your main.scss file?

**Comment by onetrev** [comment]
Hey @PixelT, @florianbouvot and @admosity ... I recently found this package https://github.com/aidistan/atom-symbols-view-plus which solves the main request of autocompletion in your HTML class declarations without having to have all the files open. Atocomplete of classes in your sass / css / etc, is still not possible as far as I know, but I think that's probably less important.

Hopefully now with the Filesystem watcher API coming in Atom 1.21, both these items can finally be tackled and with core packages! But for now at least the package above definitely helps.
